fog/ledger/enclave/api/src/lib.rs

```diff
@@ -34,28 +36,36 @@ pub type Result<T> = StdResult<T, Error>;
 /// will do the check directly.
 #[derive(Clone, Debug, Default, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
 pub struct OutputContext {
+    /// A vector for output context
     pub indexes: Vec<u64>,
+    /// A merkle root block for output context
     pub merkle_root_block: u64,
 }
 
-/// An intermediate struct for holding data required to check key images for the
-/// client. This is returned by `client_check_key_images` and allows untrusted
-/// to gather data that will be encrypted for the client in
-/// `outputs_for_client`.
-///
-/// Eventually we will do the key image check in ORAM, but for now untrusted
-/// will do the check directly.
+/// We will do the key image check in ORAM, but for now untrusted
 #[derive(Clone, Debug, Default, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
-pub struct KeyImageContext {
-    pub key_images: Vec<KeyImage>,
+pub struct UntrustedKeyImageQueryResponse {
+    /// The number of blocks at the time that the request was evaluated.
+    pub highest_processed_block_count: u64,
+
+    /// The timestamp of the highest processed block at the time that the
+    /// request was evaluated.
+    pub highest_processed_block_signature_timestamp: u64,
+
+    /// The index of the last known block, which can be obtained by calculating
+    /// last_known_block_count - 1. We don't store the index but instead store a
+    /// count so that we have a way of representing no known block (0).
+    pub last_known_block_count: u64,
```

Comment:
let's remove this also

Reply:
I would actually say that communicating it to the user would be beneficial. That way they can know the server still has catching up to do.

Reply:
yeah but we don't want to change the API right now -- we could do it in a subsequent commit